nservicebus/messaging/evolving-contracts.md

```diff
@@ -32,21 +32,86 @@ It's also possible to share messages as C# source files without packaging them i
 
 ## Common challenges
 
+### Handling breaking changes in contracts
 
-### Default values
+There are a number of approaches to handling breaking changes in messages. One of the simplest is adding a version to the type name and creating handlers for the new contract. The handlers for the old contract might still be running, to handle in-flight messages and messages generated by old endpoints.
 
-When a message is extended with additional properties, it's necessary to carefully examine what will be the default values for those properties, especially if endpoints running in other versions don't recognize them. In particular, it's important to consider how clients might interpret the default value and provide appropriate guidelines for them.
+For complex systems, especially when there are significant changes in a message type, such as adding or removing properties, or changing the property type, a step-by-step approach to evolving message contracts is recommended.
 
+#### Adding data to a contract
 
-### Handling breaking changes in contracts
+Depending on the complexity of the system, multiple options are possible. The simplest one is to:
 
-There are a number of approaches to handling breaking changes in messages. One of the simplest is adding a version to the type name and creating handlers for the new contract. The handlers for the old contract might still be running, to handle in-flight messages and messages generated by old endpoints.
+* Update the contract to the new version.
+* Update senders to use the new contract version. Ensure changes are visible for receivers, for example, by decorating the existing property with the Obsolete attribute when removing or renaming properties.
+* Update receivers to handle the new contract version. It's important that the new properties are handled correctly:
+  * It's necessary to carefully examine what the default values for the added properties will be, especially if endpoints running in other versions don't recognize them. In particular, it's important to consider how clients might interpret the default value and provide appropriate guidelines for them.
+  * Instead of relying on .NET to set the default value for int Age = 1, it's better to use nullable types and represent missing values as null.
```

Comment:
```suggestion
  * Instead of relying on .NET to set the default value for int Age = 1, it's better to use nullable types and represent missing values as null. That way, a specific check for a missing value can be added, as opposed to checking for a default value that might represent a valid value.
```

Reply:
@dvdstelt Is this better?